src/wasm-binary.h

```diff
@@ -354,14 +372,17 @@ class WasmBinaryWriter : public WasmVisitor<void> {
   }
 
   void writeMemory() {
-    o << BinaryConsts::Memory << int8_t(log2(wasm->memory.initial)) <<
-                            int8_t(log2(wasm->memory.max)) <<
-                            int8_t(1); // export memory
+    if (debug) std::cerr << "== writeMemory" << std::endl;
+    o << int8_t(BinaryConsts::Memory) << int8_t(log2(wasm->memory.initial))
+                                      << int8_t(log2(wasm->memory.max))
```

Comment:
This is using the floating-point `log2` function, but you could instead `assert(popcnt(mem) == 1);` and then use `ctz` instead of `log2`.